src/librustc_incremental/calculate_svh/svh_visitor.rs

```diff
@@ -602,7 +608,19 @@ impl<'a, 'hash, 'tcx> visit::Visitor<'tcx> for StrictVersionHashVisitor<'a, 'has
     fn visit_foreign_item(&mut self, i: &'tcx ForeignItem) {
         debug!("visit_foreign_item: st={:?}", self.st);
 
-        SawForeignItem.hash(self.st);
+        match i.node {
+            ForeignItemFn(ref fn_decl, _) => {
+                SawForeignItem(SawForeignItemComponent::Fn {
+                    variadic: fn_decl.variadic
```

Comment:
`visit_fn()` is not called for foreign items, and it's not clear that it should be (since foreign items have no body like the other cases). As far as I could tell, this is the only way to get at a foreign item's `FnDecl` without changing the visitor implementation.